firmware/application/ui/ui_freqlist.hpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2015 Jared Boone, ShareBrained Technology, Inc.
+ *
+ * This file is part of PortaPack.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#ifndef __UI_FREQLIST_H__
+#define __UI_FREQLIST_H__
+
+#include "ui.hpp"
+#include "ui_font_fixed_5x8.hpp"
+#include "ui_widget.hpp"
+#include "ui_painter.hpp"
+#include "event_m0.hpp"
+#include "message.hpp"
+#include "freqman.hpp"
+#include <cstdint>
+
+namespace ui {
+
+class FreqManUIList : public Widget {
```

Comment:
Question: how close is this to the menu widget? I wonder if it would be easier to maybe subclass and override what you need to change. I haven't actually looked into it, but it might save some code if you can.

Reply:
Honestly I don't know, my knowledge in C++ is limited.
If you find a better way, I hope to see it in a future PR :-)